### ISSUE TITLE
Fail to install on python 3.11.4 and Ubuntu 20.04

### ISSUE THREAD

**Issue by MalteMederacke** [knaidoo29/mistree#24]
Hi, 

I have issues to install mistress on python 3.11.4. On python 3.9 it works as expected. 

Installation fails with:
```
pip install mistress


Collecting mistree==1.2.0
  Using cached mistree-1.2.0.tar.gz (27 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [167 lines of output]
      /local0/tmp/pip-install-rdfn5246/mistree_698a076d5c2249449031bd19b9b3cfce/setup.py:4: DeprecationWarning:
      
        `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
        of the deprecation of `distutils` itself. It will be removed for
        Python >= 3.12. For older Python versions it will remain present.
        It is recommended to use `setuptools < 60.0` for those Python versions.
        For more details, see:
          https://numpy.org/devdocs/reference/distutils_status_migration.html
      
      
        from numpy.distutils.core import setup, Extension
      running egg_info
      running build_src
      INFO: build_src
      INFO: building extension "mistree.levy_flight.utility_random_walk" sources
      INFO: f2py options: []
      INFO: f2py:> build/src.linux-x86_64-3.11/mistree/levy_flight/utility_random_walkmodule.c
      creating build
      creating build/src.linux-x86_64-3.11
      creating build/src.linux-x86_64-3.11/mistree
      creating build/src.linux-x86_64-3.11/mistree/levy_flight
      Reading fortran codes...
          Reading file 'mistree/levy_flight/utility_random_walk.f90' (format:free)
      Post-processing...
          Block: utility_random_walk
                          Block: random_walk_fast_2d
                          Block: random_walk_fast_3d
      Applying post-processing hooks...
        character_backward_compatibility_hook
      Post-processing (stage 2)...
      Building modules...
          Building module "utility_random_walk"...
          Generating possibly empty wrappers"
          Maybe empty "utility_random_walk-f2pywrappers.f"
              Constructing wrapper function "random_walk_fast_2d"...
                x,y = random_walk_fast_2d(step_size,phi,box_size,x_start,y_start,periodic,[length])
          Generating possibly empty wrappers"
          Maybe empty "utility_random_walk-f2pywrappers.f"
              Constructing wrapper function "random_walk_fast_3d"...
                x,y,z = random_walk_fast_3d(step_size,phi,theta,box_size,x_start,y_start,z_start,periodic,[length])
          Wrote C/API module "utility_random_walk" to file "build/src.linux-x86_64-3.11/mistree/levy_flight/utility_random_walkmodule.c"
      INFO:   adding 'build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/levy_flight/fortranobject.c' to sources.
      INFO:   adding 'build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/levy_flight' to include_dirs.
      creating build/src.linux-x86_64-3.11/build
      creating build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11
      creating build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree
      creating build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/levy_flight
      copying /home/mmederacke/anaconda3/envs/segmentation/lib/python3.11/site-packages/numpy/f2py/src/fortranobject.c -> build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/levy_flight
      copying /home/mmederacke/anaconda3/envs/segmentation/lib/python3.11/site-packages/numpy/f2py/src/fortranobject.h -> build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/levy_flight
      INFO:   adding 'build/src.linux-x86_64-3.11/mistree/levy_flight/utility_random_walk-f2pywrappers.f' to sources.
      INFO: building extension "mistree.mst.utility_density" sources
      INFO: f2py options: []
      INFO: f2py:> build/src.linux-x86_64-3.11/mistree/mst/utility_densitymodule.c
      creating build/src.linux-x86_64-3.11/mistree/mst
      rmbadname1: Replacing "index" with "index_bn".
      rmbadname1: Replacing "index" with "index_bn".
      rmbadname1: Replacing "index" with "index_bn".
      rmbadname1: Replacing "index" with "index_bn".
      Reading fortran codes...
          Reading file 'mistree/mst/utility_density.f90' (format:free)
      Post-processing...
          Block: utility_density
                          Block: get_counts_2d
                          Block: get_counts_3d
                          Block: get_param_2d
                          Block: get_param_3d
      Applying post-processing hooks...
        character_backward_compatibility_hook
      Post-processing (stage 2)...
      Building modules...
          Building module "utility_density"...
          Generating possibly empty wrappers"
          Maybe empty "utility_density-f2pywrappers.f"
              Constructing wrapper function "get_counts_2d"...
                counts = get_counts_2d(x,y,dx,length,length_of_grid,[length_of_points])
          Generating possibly empty wrappers"
          Maybe empty "utility_density-f2pywrappers.f"
              Constructing wrapper function "get_counts_3d"...
                counts = get_counts_3d(x,y,z,dx,length,length_of_grid,[length_of_points])
          Generating possibly empty wrappers"
          Maybe empty "utility_density-f2pywrappers.f"
              Constructing wrapper function "get_param_2d"...
                mean_param = get_param_2d(x_param,y_param,param,dx,length,length_of_grid,[length_of_param])
          Generating possibly empty wrappers"
          Maybe empty "utility_density-f2pywrappers.f"
              Constructing wrapper function "get_param_3d"...
                mean_param = get_param_3d(x_param,y_param,z_param,param,dx,length,length_of_grid,[length_of_param])
          Wrote C/API module "utility_density" to file "build/src.linux-x86_64-3.11/mistree/mst/utility_densitymodule.c"
      INFO:   adding 'build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/mst/fortranobject.c' to sources.
      INFO:   adding 'build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/mst' to include_dirs.
      creating build/src.linux-x86_64-3.11/build/src.linux-x86_64-3.11/mistree/mst
```

I have the necessary fortran compilers installed!

Best,

Malte

**Comment by knaidoo29** [comment]
Hi @MalteMederacke , I can't quite see where/what the error is from the terminal output... is there more to the output? One issue could be causing the issue is that your `numpy` is too new? And this may cause issues with distutils compilation of fortran sources. If you are just looking to run the MST, without levy-flight constructions and density analysis you can probably bypass these issues by installing the purely python version of mistree which is located in the pure_python branch here: https://github.com/knaidoo29/mistree/tree/pure_python . You will just have to `import mistree_pp as mist` instead of `import mistree as mist`. If that fails then it's something else.

**Comment by MalteMederacke** [comment]
Hey knaidoo29,

I was very distracted by the numpy error, but it had in the end nothing to do with it. I expected it to be completely sufficient to have gcc installed, which it was not! Could fix the problem by using python 3.8.18, installing gcc with 

`conda install conda-forge::gcc` and in addition `conda install conda-forge::gcc_linux-64` and then i tried a couple of versions of `conda install conda-forge::gfortran`.

Now it runs, thank you for developing and maintaining amazing software! Much appreciated.
